optimum/exporters/utils.py

```diff
@@ -72,23 +85,40 @@
 DECODER_MERGED_NAME = "decoder_model_merged"
 
 
-def _get_submodels_for_export_stable_diffusion(
-    pipeline: "StableDiffusionPipeline",
+def _get_submodels_for_export_diffusion(
+    pipeline: "DiffusionPipeline",
 ) -> Dict[str, Union["PreTrainedModel", "ModelMixin"]]:
     """
     Returns the components of a Stable Diffusion model.
     """
-    from diffusers import StableDiffusionXLImg2ImgPipeline
 
-    models_for_export = {}
-    if isinstance(pipeline, StableDiffusionXLImg2ImgPipeline):
+    is_stable_diffusion = isinstance(
+        pipeline, (StableDiffusionPipeline, StableDiffusionImg2ImgPipeline, StableDiffusionInpaintPipeline)
+    )
+    is_stable_diffusion_xl = isinstance(
+        pipeline, (StableDiffusionXLImg2ImgPipeline, StableDiffusionXLInpaintPipeline, StableDiffusionXLPipeline)
+    )
+    is_latent_consistency_model = isinstance(
+        pipeline, (LatentConsistencyModelPipeline, LatentConsistencyModelImg2ImgPipeline)
+    )
```

Comment:
could be simplified using an `_infer_model_type_from_model_or_model_class` method and `_DIFFUSERS_TASKS_TO_PIPELINE_MAPPINGS`